index.js

```diff
@@ -1,302 +1,349 @@
-// Fix eslint shareable config (https://github.com/eslint/eslint/issues/3458)
-require('@rushstack/eslint-patch/modern-module-resolution');
+const babelEslintPlugin = require('@babel/eslint-plugin');
+const babelParser = require('@babel/eslint-parser');
+const typescriptEslintPlugin = require('@typescript-eslint/eslint-plugin');
+const typescriptParser = require('@typescript-eslint/parser');
+const eslint = require('@eslint/js');
+const eslintPluginEnact = require('eslint-plugin-enact');
+const eslintPluginJest = require('eslint-plugin-jest');
+const eslintPluginJsxA11y = require('eslint-plugin-jsx-a11y');
+const eslintPluginReact = require('eslint-plugin-react');
+const eslintPluginReactHooks = require('eslint-plugin-react-hooks');
+const eslintPluginTestingLibrary = require('eslint-plugin-testing-library');
+const globals = require('globals');
+const stylisticEslintPluginJs = require('@stylistic/eslint-plugin-js');
 
-module.exports = {
-	root: true,
-	globals: {
-		__DEV__: true,
-		process: true,
-		window: true,
-		document: true,
-		ENACT_PACK_NO_ANIMATION: true
-	},
-	env: {
-		es6: true, // Enables ES6 globals
-		'shared-node-browser': true, // restrict to common globals to preserve isomorphic support
-		commonjs: true
+const customGlobals = {
+	__DEV__: true,
+	process: true,
+	window: true,
+	document: true,
+	ENACT_PACK_NO_ANIMATION: true
+};
+
+const customTestsGlobals = {
+	'browser': true,
+	'expect': true,
+	'$': true,
+	'$$': true
+};
+
+const basicConfig = [
+	eslint.configs.recommended,
+	{
+		ignores: [
+			'**/node_modules/',
+			'**/node_modules/*',
+			'**/build/',
+			'**/build/*',
+			'**/dist/',
+			'**/dist/*',
+			'**/coverage/',
+			'**/coverage/*',
+			'**/resources/',
+			'**/resources/*'
+		]
 	},
-	extends: ['eslint:recommended', 'plugin:react/recommended', 'plugin:react-hooks/recommended'],
-	ignorePatterns: [
-		'/**/node_modules/',
-		'/**/node_modules/*',
-		'/**/build/',
-		'/**/build/*',
-		'/**/dist/',
-		'/**/dist/*',
-		'/**/coverage/',
-		'/**/coverage/*',
-		'/**/resources/',
-		'/**/resources/*'
-	],
-	parser: '@babel/eslint-parser',
-	parserOptions: {
-		ecmaFeatures: {
-			jsx: true
+	{
+		languageOptions: {
+			ecmaVersion: 'latest',
+			sourceType: 'module',
+			globals: {
+				...globals.es2015,
+				'shared-node-browser': true, // restrict to common globals to preserve isomorphic support
+				...globals.commonjs,
+				...globals.node,
```

Comment:
I have a question but, is there any reason of adding `globals.node` ?
This config file doesn't have this option before.